src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -5222,9 +5223,12 @@ public static IEnumerable<CompletionResult> CompleteVariable(string variableName
 
         private static readonly string[] s_variableScopes = new string[] { "Global:", "Local:", "Script:", "Private:" };
 
-        private static readonly char[] s_charactersRequiringQuotes = new char[] {
+        private static readonly SearchValues<char> s_charactersRequiringQuotes = SearchValues.Create(new char[] {
             '-', '`', '&', '@', '\'', '"', '#', '{', '}', '(', ')', '$', ',', ';', '|', '<', '>', ' ', '.', '\\', '/', '\t', '^',
-        };
+        });
```

Comment:
We can replace the array with string like:
```c#
SearchValues<char> s_charactersRequiringQuotes = SearchValues.Create("-`&@.........");
```

Reply:
Good catch, that is also very nice because it avoids array allocation.

Reply:
Good catch, this also avoids the allocation of the char array. Fixed in ccc336b5a089ce22a1d7a48cd37460ee32098f03.